### ISSUE TITLE
Prefer const too aggressive on destructuring

### ISSUE THREAD

**Issue by kitsonk** [denoland/deno_lint#522]
In my opinion, prefer const should not "trigger" in situations where one of the variables gets reassigned during destructuring.  For example, given:

```js
let [a, b] = ["a", "b"];
b = "c";
```

This would error that `a` is not re-assigned, but this should be allowed, as `b` gets reassigned and you can mix mutability on destructuring.

**Comment by magurotuna** [comment]
@kitsonk 
That makes sense. So you mean our `prefer-const` rule should behave like ESLint with `{ "destructuring": "all" }` enabled?

https://eslint.org/docs/rules/prefer-const#destructuring

**Comment by kitsonk** [comment]
In my opinion yes.
